ui/src/data_explorer/components/Table.js

```diff
@@ -31,50 +35,74 @@ const ChronoTable = React.createClass({
     query: shape({
       host: arrayOf(string.isRequired).isRequired,
       text: string.isRequired,
-    }),
+    }).isRequired,
     containerWidth: number.isRequired,
     height: number,
+    onEditRawStatus: func,
   },
 
   getInitialState() {
     return {
-      cellData: {
-        columns: [],
-        values: [],
-      },
+      cellData: emptyCells,
       columnWidths: {},
     }
   },
 
   getDefaultProps() {
     return {
-      height: 600,
+      height: 1000,
     }
   },
 
-  fetchCellData(query) {
+  componentDidMount() {
+    this.fetchCellData(this.props.query)
+  },
+
+  componentWillReceiveProps(nextProps) {
+    if (this.props.query.text === nextProps.query.text) {
+      return
+    }
+
+    this.fetchCellData(nextProps.query)
+  },
+
+
+  async fetchCellData(query) {
+    if (!query || !query.text) {
+      return
+    }
+
     this.setState({isLoading: true})
+    const {onEditRawStatus} = this.props
     // second param is db, we want to leave this blank
-    fetchTimeSeries(query.host, undefined, query.text).then((resp) => {
-      const cellData = _.get(resp.data, ['results', '0', 'series', '0'], false)
-      if (!cellData) {
-        return this.setState({isLoading: false})
+    try {
+      const {data} = await fetchTimeSeries(query.host, undefined, query.text)
+      this.setState({isLoading: false})
+
+      const results = _.get(data, ['results', '0'], false)
+      if (!results) {
+        return
       }
 
-      this.setState({
-        cellData,
-        isLoading: false,
-      })
-    })
-  },
+      if (_.isEmpty(results)) {
+        this.setState({cellData: emptyCells})
+        return onEditRawStatus(query.id, {warn: 'Your query is syntactically correct but returned no results'})
+      }
 
-  componentDidMount() {
-    this.fetchCellData(this.props.query)
-  },
+      const warn = _.get(results, 'error', false)
```

Comment:
Yea, this is confusing and an unfortunate part of Influx's error handling and response bodies.